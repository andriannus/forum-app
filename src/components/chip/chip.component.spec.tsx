import { beforeEach, describe, expect, it } from "vitest";
import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import Chip from "./chip.component";
import type { ChipProps } from "./chip.model";

describe("Chip component", () => {
  beforeEach(cleanup);

  it.each([{ button: true }, { small: true }, { button: true, small: true }])(
    "should match snapshot with various kinds of props",
    async (props: Partial<ChipProps>) => {
      render(
        <Chip data-testid="chip" {...props}>
          Hello world
        </Chip>,
      );

      const component = screen.queryByTestId("chip");

      expect(component).toMatchSnapshot();
    },
  );

  it("should contain `is-active` when active prop is `true`", () => {
    render(
      <Chip data-testid="chip" active>
        Hello world
      </Chip>,
    );

    const component = screen.queryByTestId("chip");

    expect(component).toHaveClass("is-active");
  });

  it("should contain `Chip--small` when small prop is `true`", () => {
    render(
      <Chip data-testid="chip" small>
        Hello world
      </Chip>,
    );

    const component = screen.queryByTestId("chip");

    expect(component).toHaveClass("Chip--small");
  });
});
