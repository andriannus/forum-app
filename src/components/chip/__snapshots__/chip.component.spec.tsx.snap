// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Chip component > should match snapshot with various kinds of props 1`] = `
<div
  class="Chip Chip--button"
  data-testid="chip"
>
  Hello world
</div>
`;

exports[`Chip component > should match snapshot with various kinds of props 2`] = `
<div
  class="Chip Chip--small"
  data-testid="chip"
>
  Hello world
</div>
`;

exports[`Chip component > should match snapshot with various kinds of props 3`] = `
<div
  class="Chip Chip--button Chip--small"
  data-testid="chip"
>
  Hello world
</div>
`;
